src/backend/cdb/cdbmutate.c

```diff
@@ -1903,23 +1908,25 @@ shareinput_walker(Node *node, ShareInputContext *ctx)
 
 			if (nl->join.prefetch_inner)
 			{
-				plan->lefttree = save_righttree;
-				plan->righttree = save_lefttree;
+				nl->join.plan.lefttree = save_righttree;
```

Comment:
if in this case `nl->join.plan == ((Plan *) node)` ?
or it is another plan?

Reply:
`nl->join.plan` and `((Plan *) node)` are exactly the same.

Reply:
what this code does?

Reply:
Changes the right and left parts of join.

Reply:
are they not same?

Reply:
No, they not. This is why this logic is here.

Reply:
where they changes in `nl->join.plan` ?

Reply:
Can't understand the question. Can you reformulate it?

Reply:
I do not understand `why nl->join.plan.lefttree != save_righttree`?

Reply:
Because we copied a pointer to `save_righttree` from `((Plan *) node)->righttree`, not from `((Plan *) node)->lefttree`.

Reply:
OK. I see now